### PR TITLE
Optional results for oneAPI DBSCAN

### PR DESCRIPTION
**Pull request by amgrigoriev**

### CODE REVIEW

File: cpp/oneapi/dal/algo/dbscan/common.hpp [hunk 1]
```diff
@@ -48,6 +49,34 @@ using v1::by_default;
...
+} // namespace detail
+
+/// Result options are used to define
+/// what should algorithm return
+namespace result_options {
+
+const result_option_id responses = detail::get_responses_id();
+const result_option_id core_observation_indices = detail::get_core_observation_indices_id();
+const result_option_id core_observations = detail::get_core_observations_id();
+const result_option_id core_flags = detail::get_core_flags_id();
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/dbscan/common.hpp:76]: It's wrong! Please mark them as `const inline`

File: cpp/oneapi/dal/algo/dbscan/compute_types.cpp [hunk 2]
```diff
@@ -74,21 +76,37 @@ compute_result<Task>::compute_result() : impl_(new compute_result_impl<Task>{})
 
 template <typename Task>
 const table& compute_result<Task>::get_responses() const {
+    using msg = dal::detail::error_messages;
+    if (!bool(get_result_options() & result_options::responses)) {
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/dbscan/compute_types.cpp:80]: Please use `.test(...)`

Comment (KulikovNikita) [reply cpp/oneapi/dal/algo/dbscan/compute_types.cpp:80]:
```suggestion
    if (!get_result_options().test(result_options::responses)) {
```

File: cpp/oneapi/dal/algo/dbscan/test/batch.cpp [hunk 3]
```diff
@@ -30,14 +30,19 @@ namespace oneapi::dal::dbscan::test {
 namespace te = dal::test::engine;
 namespace la = te::linalg;
 
+constexpr inline std::uint64_t mask_full = 0xffffffffffffffff;
+
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/dbscan/test/batch.cpp:34]: What for?

File: cpp/oneapi/dal/algo/dbscan/test/batch.cpp [hunk 4]
```diff
@@ -100,10 +105,68 @@ class dbscan_batch_test : public te::float_algo_fixture<std::tuple_element_t<0,
...
 
 using dbscan_types = COMBINE_TYPES((float, double), (dbscan::method::brute_force));
 
+TEMPLATE_LIST_TEST_M(dbscan_batch_test,
+                     "dbscan compute mode check",
+                     "[dbscan][batch]",
+                     dbscan_types) {
+    using Float = std::tuple_element_t<0, TestType>;
+
+    Float data[] = { 0.0, 5.0, 0.0, 0.0, 0.0, 1.0, 1.0, 4.0, 0.0, 0.0, 1.0, 0.0, 0.0, 5.0, 1.0 };
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/dbscan/test/batch.cpp:153]: What about making it `constexpr`?

File: cpp/oneapi/dal/algo/dbscan/test/batch.cpp [hunk 5]
```diff
@@ -100,10 +105,68 @@ class dbscan_batch_test : public te::float_algo_fixture<std::tuple_element_t<0,
...
+    }
 };
 
 using dbscan_types = COMBINE_TYPES((float, double), (dbscan::method::brute_force));
 
+TEMPLATE_LIST_TEST_M(dbscan_batch_test,
+                     "dbscan compute mode check",
+                     "[dbscan][batch]",
+                     dbscan_types) {
+    using Float = std::tuple_element_t<0, TestType>;
```
Comment (KulikovNikita) [inline cpp/oneapi/dal/algo/dbscan/test/batch.cpp:151]: Please use `float_t` instead. We have an agreement that only type in template parameters should be called in Camel Case. In all other cases it should be done in snake case like `float_t`

File: cpp/oneapi/dal/algo/dbscan/test/batch.cpp (hunk 3 above)
Comment (amgrigoriev) [reply cpp/oneapi/dal/algo/dbscan/test/batch.cpp:34]: It is used at line 159 to check how it works with all bits set

File: cpp/oneapi/dal/algo/dbscan/common.hpp (hunk 1 above)
Comment (amgrigoriev) [reply cpp/oneapi/dal/algo/dbscan/common.hpp]: done

File: cpp/oneapi/dal/algo/dbscan/test/batch.cpp (hunk 5 above)
Comment (amgrigoriev) [reply cpp/oneapi/dal/algo/dbscan/test/batch.cpp]: Done

**Pull request merged**